packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -1010,45 +900,150 @@ private boolean updateMapStyle(String style) {
+    if (googleMap == null) {
+      mapReadyResult = result;
+    } else {
+      result.success();
+    }
+  }
+
+  @Override
+  public @NonNull Messages.PlatformPoint getScreenCoordinate(
+      @NonNull Messages.PlatformLatLng latLng) {
+    if (googleMap == null) {
+      throw new FlutterError(
+          "GoogleMap uninitialized",
+          "getScreenCoordinate called prior to map initialization",
+          null);
+    }
+    Point screenLocation =
+        googleMap.getProjection().toScreenLocation(Convert.latLngFromPigeon(latLng));
+    return Convert.pointToPigeon(screenLocation);
+  }
+
+  @Override
+  public @NonNull Messages.PlatformLatLng getLatLng(
+      @NonNull Messages.PlatformPoint screenCoordinate) {
+    if (googleMap == null) {
+      throw new FlutterError(
+          "GoogleMap uninitialized", "getLatLng called prior to map initialization", null);
+    }
+    LatLng latLng =
+        googleMap.getProjection().fromScreenLocation(Convert.pointFromPigeon(screenCoordinate));
+    return Convert.latLngToPigeon(latLng);
+  }
+
+  @Override
+  public @NonNull Messages.PlatformLatLngBounds getVisibleRegion() {
+    if (googleMap == null) {
+      throw new FlutterError(
+          "GoogleMap uninitialized", "getVisibleRegion called prior to map initialization", null);
+    }
+    LatLngBounds latLngBounds = googleMap.getProjection().getVisibleRegion().latLngBounds;
+    return Convert.latLngBoundsToPigeon(latLngBounds);
+  }
+
+  @Override
+  public @NonNull Double getZoomLevel() {
+    if (googleMap == null) {
+      throw new FlutterError(
+          "GoogleMap uninitialized", "getZoomLevel called prior to map initialization", null);
+    }
+    return (double) googleMap.getCameraPosition().zoom;
+  }
+
+  @Override
+  public void showInfoWindow(@NonNull String markerId) {
+    markersController.showMarkerInfoWindow(markerId);
+  }
+
+  @Override
+  public void hideInfoWindow(@NonNull String markerId) {
+    markersController.hideMarkerInfoWindow(markerId);
+  }
+
+  @NonNull
+  @Override
+  public Boolean isInfoWindowShown(@NonNull String markerId) {
+    return markersController.isInfoWindowShown(markerId);
+  }
+
+  @Override
+  public @NonNull Boolean setStyle(@NonNull String style) {
+    return updateMapStyle(style);
+  }
+
+  @Override
+  public @NonNull Boolean didLastStyleSucceed() {
+    return lastSetStyleSucceeded;
+  }
+
+  @Override
+  public void clearTileCache(@NonNull String tileOverlayId) {
+    tileOverlaysController.clearTileCache(tileOverlayId);
+  }
+
+  @Override
+  public void takeSnapshot(@NonNull Messages.Result<byte[]> result) {
+    if (googleMap == null) {
+      result.error(new FlutterError("GoogleMap uninitialized", "takeSnapshot", null));
+    } else {
+      googleMap.snapshot(
+          bitmap -> {
+            if (bitmap == null) {
+              result.error(new FlutterError("Snapshot failure", "Unable to take snapshot", null));
```

Comment:
This is another warning I added handling for; `bitmap` can explicitly be null in the API docs, but that wasn't handled.